mesonbuild/compilers/c.py

```diff
@@ -678,7 +678,8 @@ def has_function(self, funcname, prefix, env, extra_args=None, dependencies=None
             head, main = self._no_prototype_templ()
         templ = head + stubs_fail + main
 
-        if self.links(templ.format(**fargs), env, extra_args, dependencies):
+        if not funcname.startswith('__builtin_') and \
```

Comment:
I do not thing meson should have this assumption, why is this needed? Optimization?

Reply:
Yes, that check will always fail with builtins because it tries to take the address of the function.

Reply:
Which is fine, it just fails and meson goes on to the builtin test.
But what happens if there is a function that matches that pattern that is not a builtin? I think hardcoding that naming convention is just assuming things that might not be true. So i would always let that test fail. And if in some odd environment a function with such an name is an actual external function it will not break just because of naming.